### ISSUE TITLE
Major issue: gr.Dropdown() has weird behavior with tuples

### ISSUE THREAD

**Issue by sbarman25** [gradio-app/gradio#6707]
### Describe the bug

Hi,

As we know, it's possible to pass tuples (name, value) as choices in dropdown component.

If we select an option, then we open the dropdown list again, and click outside dropdown, it passes the `name` instead of `value`, then removes the selected option from display. Please see the video for complete details.

Expected behavior: 
1) It should retain the previously selected value
2) Should pass `value` instead of `name`

Thanks.

### Have you searched existing issues?  🔎

- [X] I have searched and found no existing issues

### Reproduction

```
import gradio as gr


choices = [
    ("apple", 1),
    ("banana", 2),
    ("orange", 3),
    ("pear", 4),
    ("strawberry", 5)
]

with gr.Blocks() as demo:
    drpDown = gr.Dropdown(choices=choices, label="Fruit", value=1, interactive=True)

    @drpDown.input(inputs=[drpDown], outputs=None)
    def handle_dropdown_change(value):
        gr.Info(str(value))

demo.queue(default_concurrency_limit=13).launch(server_name='0.0.0.0', server_port=7860)
```

### Screenshot

https://github.com/gradio-app/gradio/assets/14152255/d3766079-3db3-45bc-a896-d9ac3d6bd82b

### System Info

```shell
gradio==4.7.1
```

### Severity

Blocking usage of gradio

**Comment by freddyaboulton** [comment]
HI @sbarman-mi9 ! This got fixed yesterday in #6694 . Will be out in the next release!

**Comment by sbarman-mi9** [comment]
Thanks @freddyaboulton
